keycloak-event-emitter-tests/src/test/java/io/cloudtrust/keycloak/kafkaeventemitter/SerializationUtilsTest.java

```diff
@@ -0,0 +1,289 @@
+        ByteBuffer buffer = io.cloudtrust.keycloak.kafkaeventemitter.SerialisationUtils.toFlat(adminEvent);
+        flatbuffers.events.AdminEvent deserializedAdminEvent = flatbuffers.events.AdminEvent.getRootAsAdminEvent(buffer);
+        Assert.assertTrue(equals(adminEvent, deserializedAdminEvent));
+        Assert.assertEquals(UID, deserializedAdminEvent.uid());
+    }
+
+    private Event createEvent() {
+        Event event = new Event();
+        event.setTime(120000);
+        event.setType(EventType.CLIENT_LOGIN);
+        event.setRealmId("realmId");
+        event.setClientId("clientId");
+        event.setUserId("userId");
+        event.setSessionId("sessionId");
+        event.setIpAddress("127.0.0.1");
+        event.setError("Error");
+
+        Map<String, String> details = new HashMap<>();
+        details.put("detailsKey1", "detailValue1");
+        details.put("detailsKey2", "detailsValue2");
+        event.setDetails(details);
+        return event;
+    }
+
+    private Event createMinimalEvent() {
+        Event event = new Event();
+        event.setTime(120000);
+        event.setType(EventType.CLIENT_LOGIN);
+        return event;
+    }
+
+    private ExtendedAdminEvent createExtendedAdminEvent() {
+        IdentifiedAdminEvent iae = new IdentifiedAdminEvent(UID, createAdminEvent());
+        ExtendedAdminEvent eae = new ExtendedAdminEvent(iae);
+        eae.getAuthDetails().setUsername("authDetails-username");
+        eae.getDetails().put("user_id", "userid");
+        eae.getDetails().put("username", "username");
+        return eae;
+    }
+
+    private AdminEvent createAdminEvent() {
+        AdminEvent adminEvent = new AdminEvent();
+        adminEvent.setTime(120000);
+        adminEvent.setRealmId("realmId");
+        AuthDetails authDetails = new AuthDetails();
+        authDetails.setRealmId("authDetails-realmId");
+        authDetails.setClientId("authDetails-clientId");
+        authDetails.setIpAddress("authDetails-ipAddress");
+        authDetails.setUserId("authDetails-userId");
+        adminEvent.setAuthDetails(authDetails);
+        adminEvent.setResourceType(ResourceType.AUTHORIZATION_RESOURCE);
+        adminEvent.setOperationType(OperationType.CREATE);
+        adminEvent.setResourcePath("resource/path");
+        adminEvent.setRepresentation("representation");
+        adminEvent.setError("error");
+        return adminEvent;
+    }
+
+    private ExtendedAdminEvent createMinimalExtendedAdminEvent() {
+        IdentifiedAdminEvent iae = new IdentifiedAdminEvent(UID, createMinimalAdminEvent());
+        return new ExtendedAdminEvent(iae);
+    }
+
+    private AdminEvent createMinimalAdminEvent() {
+        AdminEvent adminEvent = new AdminEvent();
+        adminEvent.setTime(120000);
+        return adminEvent;
+    }
+
+    private boolean equals(Event event, flatbuffers.events.Event eventFlat) {
+        if (event.getTime() != eventFlat.time()) {
```

Comment:
You can use EqualsBuilder in apache.commons-lang3

Reply:
I did not change anything to serialization